File: [Beecrowd]/ad-hoc/1129.py
while(True):
    n = int(input())
    if (n == 0): break
    for _ in range(n):
        respostas = list(map(int,input().split()))

        achou = False
        res = -1

        for i,j in enumerate(respostas):
            if (j<=127):
                if (not achou):
                    res = i
                    achou = True
                else:
                    res = -1
                    break
        match res:
            case 0: print("A")
            case 1: print("B")
            case 2: print("C")
            case 3: print("D")
            case 4: print("E")
            case -1: print("*")

File: [Beecrowd]/ad-hoc/1129.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
    int n = -1;
    while (n != 0){
        cin >> n;
        if (n==0) return 0;

        for (int i = 0; i<n;i++){
            int a,b,c,d,e;
            cin >> a >> b >> c >> d >> e;
            vector<int> respostas = {a,b,c,d,e};

            bool achou = false;
            int res = -1;

            for (int j = 0; j<5; j++){
                if(respostas[j]<=127){
                    if(not achou){
                        res = j;
                        achou = true;
                    }else{
                        res = -1;
                        break;
                    }
                }
            }
            string letras = "ABCDE";
            if (res == -1) cout << "*" << endl;
            else cout << letras[res] << endl;
        }
    }
    return 0;
}